tests/Unit/Evolution/DgSubcell/Test_Mesh.cpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "Evolution/DgSubcell/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace {
+template <Spectral::Basis BasisType, Spectral::Quadrature QuadratureType>
+void test_mesh() noexcept {
+  constexpr size_t min_num_pts =
+      Spectral::minimum_number_of_points<BasisType, QuadratureType>;
+  constexpr size_t max_num_pts = Spectral::maximum_number_of_points<BasisType>;
+  for (size_t i = min_num_pts; i < max_num_pts; ++i) {
+    CHECK(evolution::dg::subcell::fd::mesh(
+              Mesh<1>(i, BasisType, QuadratureType)) ==
+          Mesh<1>{2 * i - 1, Spectral::Basis::FiniteDifference,
+                  Spectral::Quadrature::CellCentered});
+    CHECK(evolution::dg::subcell::fd::mesh(
+              Mesh<2>(i, BasisType, QuadratureType)) ==
+          Mesh<2>{2 * i - 1, Spectral::Basis::FiniteDifference,
+                  Spectral::Quadrature::CellCentered});
+    CHECK(evolution::dg::subcell::fd::mesh(
+              Mesh<3>(i, BasisType, QuadratureType)) ==
+          Mesh<3>{2 * i - 1, Spectral::Basis::FiniteDifference,
+                  Spectral::Quadrature::CellCentered});
+  }
```

Comment:
I think it might be nice to test using a mesh with non-uniform extents.

One easy way to at least hit this case without making the test much more complicated would be to add, after the for loop, one check each in 2D and 3D, perhaps like `Mesh<2>({{5,8}}, ...` and `Mesh<3>({{9,8,11}}, ...`